wuvt/admin/auth/views.py

```diff
@@ -0,0 +1,242 @@
+from flask import abort, flash, jsonify, make_response, redirect, \
+    render_template, request, url_for
+from flask_login import current_user
+
+from wuvt import app, auth_manager, db
+from wuvt.admin import bp
+from wuvt.auth.models import User, UserRole, GroupRole
+
+
+@bp.route('/roles/users/add', methods=['GET', 'POST'])
+@auth_manager.check_access('admin')
+def role_add_user():
+    error_fields = []
+
+    if request.method == 'POST':
+        role = request.form['role']
+        if not role in auth_manager.all_roles:
+            error_fields.append('role')
+
+        user = User.query.get(request.form['user'])
+        if user is None:
+            error_fields.append('user')
+
+        if len(error_fields) <= 0:
+            user_id = int(request.form['user'])
+
+            existing = UserRole.query.filter_by(
+                user_id=user_id, role=role).count()
+            if existing > 0:
+                flash("That role was already assigned to that user.")
+            else:
+                db.session.add(UserRole(user_id, role))
+                db.session.commit()
+
+                flash("The role has been assigned to the user.")
+
+            return redirect(url_for('.roles'), 303)
+
+    users = User.query.order_by('name').all()
+
+    return render_template('admin/role_add_user.html', users=users,
+                           roles=sorted(auth_manager.all_roles),
+                           error_fields=error_fields)
+
+
+@bp.route('/roles/users/remove/<int:id>', methods=['POST', 'DELETE'])
+@auth_manager.check_access('admin')
+def role_remove_user(id):
+    user_role = UserRole.query.get_or_404(id)
+    db.session.delete(user_role)
+    db.session.commit()
+
+    if request.method == 'DELETE' or request.wants_json():
+        return jsonify({
+            '_csrf_token': app.jinja_env.globals['csrf_token'](),
+        })
+    else:
+        return redirect(url_for('.roles', 303))
+
+
+@bp.route('/roles/groups/add', methods=['GET', 'POST'])
```

Comment:
Minor nitpick - the API syntax is kind of awkward, can we make this choose to add/remove based on method to /roles/groups (PUT/DELETE) instead?